backend/src/apiserver/template/v2_template.go

```diff
@@ -306,3 +311,55 @@ func IsPlatformSpecWithKubernetesConfig(template []byte) bool {
 	_, ok := platformSpec.Platforms["kubernetes"]
 	return ok
 }
+
+func (t *V2Spec) validatePipelineJobInputs(job *pipelinespec.PipelineJob) error {
+	// The pipeline requires no input
+	if t.spec.GetRoot().GetInputDefinitions() == nil {
+		return nil
+	}
+
+	runtimeConfig := job.GetRuntimeConfig()
+	if runtimeConfig == nil {
+		return fmt.Errorf("pipeline requiring input has empty runtime config")
+	}
+
+	for name, param := range t.spec.GetRoot().GetInputDefinitions().GetParameters() {
+		if input, ok := runtimeConfig.GetParameterValues()[name]; !ok {
+			// If the parameter is optional, or there is a default value, it's ok to not have a user input
+			if !param.GetIsOptional() && param.GetDefaultValue() == nil {
+				return fmt.Errorf("parameter %s is not optional, yet has neither default value nor user provided value", name)
+			}
+		} else {
+			// Verify the parameter type is correct
+			switch param.GetParameterType() {
+			case pipelinespec.ParameterType_PARAMETER_TYPE_ENUM_UNSPECIFIED:
+				return fmt.Errorf("input parameter %s has unspecified type", name)
+			case pipelinespec.ParameterType_NUMBER_DOUBLE, pipelinespec.ParameterType_NUMBER_INTEGER:
+				if _, ok := input.GetKind().(*structpb.Value_NumberValue); !ok {
```

Comment:
I wonder if we can actually get the kind of an input value. Can you test this via UI, when you put a number in the input box, do this get numbervalue here or stringvalue?

Reply:
Thank you, I just verified it.